hoomd/mpcd/RandomSubsetPicker.cc

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2009-2019 The Regents of the University of Michigan
+// This file is part of the HOOMD-blue project, released under the BSD 3-Clause License.
+
+// Maintainer: mphoward
+
+#include "RandomSubsetPicker.h"
+
+/* 
+ * This file contains some definitions of functions used in RandomSubsetPicker::operator()
+ * see RandomSubsetPicker.h file for declarations for RandomSubsetPicker and definition of operator()
+ */
+void mpcd::RandomSubsetPicker::assignPicks(GPUArray<unsigned int>& picks, unsigned int& N_pick)
+    {
+    /* 
+     * The function stores the indexes of picked particles(indices of picked particles according to the *original* flags array) in picks
+     * and number of picked particles in N_pick.
+     * This chooses whether to take CPU path or GPU path according to m_exec_conf
+     *
+     * \param picks Indices of picked particles in \a flags 
+     * \param m_Npick number of picked particles 
+     * \param m_picks_idx Indices of picked particles in \a m_flags_idx array
+     * \param m_flags_idx Indices of particles in \a flags which are set (1=set)
+     * \param N_pick number of picked particles which should be set to m_Npick in this function
+     */
```

Comment:
This is a doxygen comment, it needs to go above the function (at the same level of indentation), and the first line of the comment should be `/*!`.